src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java

```diff
@@ -177,29 +180,84 @@ public void execute()
             }
             else
             {
-                DEPLOYREQUESTS.add( pdr );
-                addedDeployRequest = true;
+                if ( altReleaseDeploymentRepository != null )
+                {
+                    getPluginContext().put(
+                        DEPLOY_ALT_RELEASE_DEPLOYMENT_REPOSITORY,
+                        altReleaseDeploymentRepository
+                    );
+                }
+                if ( altSnapshotDeploymentRepository != null )
+                {
+                    getPluginContext().put(
+                        DEPLOY_ALT_SNAPSHOT_DEPLOYMENT_REPOSITORY,
+                        altSnapshotDeploymentRepository
+                    );
+                }
+                if ( altDeploymentRepository != null )
+                {
+                    getPluginContext().put(
+                        DEPLOY_ALT_DEPLOYMENT_REPOSITORY,
+                        altDeploymentRepository
+                    );
+                }
```

Comment:
If we can add null value - I would not check for null

Reply:
I still think that checking for null is not necessary.

Reply:
w/o nullcheck:
```
Caused by: java.lang.NullPointerException
    at java.util.concurrent.ConcurrentHashMap.putVal (ConcurrentHashMap.java:1011)
    at java.util.concurrent.ConcurrentHashMap.put (ConcurrentHashMap.java:1006)
    at org.apache.maven.plugins.deploy.DeployMojo.execute (DeployMojo.java:183)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)

```
You know, that plugin context is not a simple Map...